llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -1664,7 +1668,6 @@ bool PreRARematStage::allUsesAvailableAt(const MachineInstr *InstToRemat,
         if (LM.none())
           break;
       }
-      assert(LM.none());
```

Comment:
Testing had issues with this assert -- would like to understand why -- looking into it.

Reply:
We may have non empty LM if our register does not have a subrange for every subregister. I've attached a reproducer of this condition (admittedly, it looks a bit funky but it represents a case in hipBlender).

If a superregister has undef subregisters, LIS won't produce a subrange for those subregisters. Thus, when we walk over the subranges for a full register use, the sum of the subrange masks will not be the full register mask.